src/protocol/invite.rs

```diff
@@ -42,36 +42,43 @@ pub(crate) struct InvitationType(pub libparsec::types::InvitationType);
 
 #[pymethods]
 impl InvitationType {
-    #[new]
-    fn new(invitation_str: &str) -> PyResult<Self> {
-        match invitation_str {
-            "DEVICE" => Ok(Self(libparsec::types::InvitationType::Device)),
-            "USER" => Ok(Self(libparsec::types::InvitationType::User)),
-            _ => Err(PyNotImplementedError::new_err("")),
-        }
-    }
-
-    #[classmethod]
+    #[classattr]
     #[pyo3(name = "DEVICE")]
-    fn device(_cls: &PyType) -> Self {
-        Self(libparsec::types::InvitationType::Device)
+    fn device() -> &'static PyObject {
+        lazy_static::lazy_static! {
+            static ref VALUE: PyObject = {
+                Python::with_gil(|py| {
+                    InvitationType(libparsec::types::InvitationType::Device).into_py(py)
+                })
+            };
+        };
+        &VALUE
```

Comment:
I would like to say it's overkill to use lazy_static there, It's like a number

Reply:
Nop because without this lazy_static, we would return a different object each time we do `InvitationType.USER`, which means `assert InvitationType.USER is InvatitionType.USER` would fail.
This is a bad idea because it would create hard to track bug when using `is` operator instead of `==` by mistake ;-)

Reply:
Given we have already the GIL, we could use a static variable without relying on lazy_static (i.e. we would use the python mutex to protect our own static variable), however I found using lazy_static just simpler to implement (no risk of hidden bug because we removed the `py` from the function signature for whatever reason and now the static is subject to access concurrency)